onnxruntime/core/providers/nnapi/nnapi_builtin/builders/model_builder.h

```diff
@@ -156,19 +144,17 @@ class ModelBuilder {
 
   std::unordered_set<std::string> unique_names_;
 
-  TargetDeviceOption target_device_option_{TargetDeviceOption::ALL_DEVICES};
-  std::vector<ANeuralNetworksDevice*> nnapi_target_devices_;
-  std::string nnapi_target_devices_detail_;  // Debug info for target devices
+  const std::vector<DeviceWrapper>& nnapi_target_devices_;
```

Comment:
would it be simpler if ModelBuilder owns the vector? otherwise consider `gsl::span<const DeviceWrapper>`

Reply:
It seems better to have EP owning the devices vector, becuase EP created it and used it for selecting ops.